images/dvcr-artifact/pkg/registry/registry.go

```diff
@@ -105,6 +106,95 @@ func NewDataProcessor(ds datasource.DataSourceInterface, dest DestinationRegistr
 	}, nil
 }
 
+func createImgFromDevice(deviceName, imgName string) error {
+	blockDevice, err := os.Open(deviceName)
+	if err != nil {
+		fmt.Errorf("can not get open device")
+		return err
+	}
+	defer blockDevice.Close()
+
+	imgFile, err := os.Create(imgName)
+	if err != nil {
+		fmt.Errorf("can not get create image")
+		return err
+	}
+	defer imgFile.Close()
+
+	buf := make([]byte, 1024*1024)
+	_, err = io.CopyBuffer(imgFile, blockDevice, buf)
+	if err != nil {
+		fmt.Errorf("can not get copy to image")
+		return err
+	}
+
+	return nil
+}
+
+func (p DataProcessor) ProcessFromBlockDevice(ctx context.Context) (ImportRes, error) {
+	blockDevicePath := "/dev/xvda"
+	uuid, err := uuid.NewUUID()
+	imgPath := "/tmp/" + uuid.String() + ".raw"
+
```

Comment:
Seems incorrect. No need to copy into /tmp and eat local disk storage. `imgPath := "/dev/xvda"` is totally OK.

Reply:
Yes. Fixed